sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSortRel.java

```diff
@@ -154,22 +155,29 @@ public int getCount() {
           BeamIOSinkRel.class.getSimpleName(),
           pinput);
       PCollection<Row> upstream = pinput.get(0);
-      Type windowType =
-          upstream.getWindowingStrategy().getWindowFn().getWindowTypeDescriptor().getType();
-      if (!windowType.equals(GlobalWindow.class)) {
-        throw new UnsupportedOperationException(
-            "`ORDER BY` is only supported for GlobalWindow, actual window: " + windowType);
-      }
 
-      BeamSqlRowComparator comparator =
-          new BeamSqlRowComparator(fieldIndices, orientation, nullsFirst);
+      // There is a need to separate ORDER BY LIMIT and LIMIT:
+      //  - GroupByKey (used in Top) is not allowed on unbounded data in global window so ORDER BY ... LIMIT
+      //    works only on bounded data.
+      //  - Just LIMIT operates on unbounded data, but across windows.
+      if (fieldIndices.size() == 0) {
+        return upstream
+            .apply(Window.into(new GlobalWindows()))
```

Comment:
Nit: let's have a TODO here to remind that we might need this fixed global window strategy to a user defined one?

Reply:
Done